pulp_smash/pulp3/utils.py

```diff
@@ -131,6 +133,33 @@ def download_content_unit(_cfg, distribution, unit_path, **kwargs):
     return utils.http_get(unit_url, **kwargs)
 
 
+def download_content_unit_return_requests_response(_cfg, distribution, unit_path, **kwargs):
```

Comment:
I feel like we should decompose this into two functions, one that produces a pulp url for a content unit, and one that makes the request.

Then the url logic can be shared between these two download functions, and it'll be easier to phase the other function out (since once the url logic is removed it's just a thin wrapper around `http_get`)